### PR TITLE
fix: return a random Symbol from dev hook to stop exponential backoff

### LINKED ISSUE #484: [bug]: 6.0.2 retries successful uploads
### Provide environment information

```bash
System:
    OS: macOS 14.1.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 24.15 GB / 64.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 19.4.0 - ~/.nvm/versions/node/v19.4.0/bin/node
    Yarn: 3.6.1 - ~/.yarn/bin/yarn
    npm: 9.2.0 - ~/.nvm/versions/node/v19.4.0/bin/npm
  Browsers:
    Chrome: 112.0.5615.137
    Safari: 17.1
  npmPackages:
    @uploadthing/react: ^6.0.1 => 6.0.1 
    typescript: ^4.3.5 => 4.9.5 
    uploadthing: ^6.0.2 => 6.0.2
```

### Describe the bug

image

Images are getting uploaded successfully after updating to 6.0.2, but uploadthing seems to believe they're failing so it keeps retrying.

### Link to reproduction

(I haven't used stackblitz before, but the link is throwing an error)

### To reproduce

I'm using basically the default setup. Here's my fileRoute:

```
recipeImage: f({ image: { maxFileSize: '4MB' } }).onUploadComplete(
    async ({ metadata, file }) => {
      console.log('uploaded:', file.url, metadata)
    }
  ),
```

### Additional information

_No response_

### 👨‍👧‍👦 Contributing

- [ ] 🙋‍♂️ Yes, I'd be down to file a PR fixing this bug!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by juliusmarminge**
Closes #484

### CODE REVIEW

**Comment by juliusmarminge** [comment]
Current dependencies on/for this PR:
* main
  * **PR #487** <a href="https://app.graphite.dev/github/pr/pingdotgg/uploadthing/487?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/pingdotgg/uploadthing/487?utm_source=stack-comment).

**Review by markflorkowski — APPROVED** [review]

**Review by markflorkowski — APPROVED** [review]

File: packages/shared/src/utils.ts [hunk 1]
```diff
@@ -177,11 +183,13 @@ export async function pollForFileData(
       console.error(
         `[UT] Error polling for file data for ${opts.url}: ${maybeJson.message}`,
       );
-      return undefined;
+      return null;
```
Comment (juliusmarminge) [inline packages/shared/src/utils.ts:186]: am i right here @markflorkowski ?

Comment (markflorkowski) [reply packages/shared/src/utils.ts:186]: seems correct.

Comment (markflorkowski) [reply packages/shared/src/utils.ts:186]: Perhaps we should actually throw here?

**Pull request merged**